version: "3.5"

services:
  bima360_app:
    container_name: bima360_app
    build:
      context: .
    ports:
      - "8500:8000"
    volumes:
      - ./bima360_static_file/:/app/media/
    command: >
      sh -c "
             chmod -R 777 /app/media &&
             chown -R 1000:1000 /app/media &&
             python manage.py wait_for_db && 
             python manage.py migrate &&
             python manage.py runserver 0.0.0.0:8000"
    env_file:
      - app/dev.env
    environment:
      - DB_HOST=bima360_db
      - DJANGO_ENV=prod
    depends_on:
      - bima360_db
    networks:
      - bima360-network

  bima360_db:
    container_name: bima360_db
    image: postgres:15
    env_file:
      - app/dev.env
    ports:
      - "5432:5432"
    volumes:
      - ./bima360_db/:/var/lib/postgresql/data/
    networks:
      - bima360-network

  bima360_db_admin:
    container_name: bima360_db_admin
    image: dpage/pgadmin4
    restart: always
    volumes:
      - ./bima360_pgadmin/:/var/lib/pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@bima360.com
      PGADMIN_DEFAULT_PASSWORD: bima_admin_pwd
    ports:
      - "8501:80"
    networks:
      - bima360-network

  celery:
    build: .
    command: python -m celery -A app worker -l info -E --pool=solo
    volumes:
      - .:/app


networks:
  bima360-network:
    name: bima360_network
    driver: bridge